vignettes/example.Rmd

```diff
@@ -0,0 +1,276 @@
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# installation
+
+```{r load libraries}
+library(plant)
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(patchwork)
+```
+
+Activate logger
+
+```{r}
+plant_log_console()
+```
+
+Setting some base parameters.
+
+```{r}
+base_parameters <- function() {
+  p0 <- scm_base_parameters("FF16", "FF16_Env")
+
+  p0$strategy_default$lma <- 0.07
+  p0$strategy_default$hmat <- 15
+  p0$strategy_default$rho <- 700
+  p0$strategy_default$a_l1 <- 2.17
+  p0$strategy_default$a_l2 <- 0.5
+  
+  p0
+}
+```
+
+We can now create a function called `run_mypatch` which takes a number of a arguments, including `traits` which we assign a trait matrix. The `expand_parameters` function takes this trait matrix as an argument which allows us to simulate patches with multiple functional strategis, in addition to the base_parameters and any newly defined hyperparameters such as B_lf1. We also provide a latitude as an argument, as well as whether or not we want to run the patch to equilibrium seed rain.
+
+```{r}
+run_mypatch <- function(
+  traits =  trait_matrix(c(0.07), c("lma")),
+  B_lf1 = 1,
+  seed_rain = 100,
+  p0 = base_parameters(),
+  find_equilbrium = FALSE,
+  latitude = 28.182
+  ) {
+  
+  hyper_par_fn = make_FF16_hyperpar(B_lf1 = B_lf1, latitude = latitude)
+  p1 <- expand_parameters(traits, p0, hyper_par_fn, mutant = FALSE)
+  p1$seed_rain <- seed_rain
+  
+  if(find_equilbrium)
+    result <- equilibrium_seed_rain(p1)
+  else
+    result <- build_schedule(p1)
+  
+  # gather outputs at each time step
+  run_scm_collect(result)
+}
+```
+
+
+Run patch for two species.  Here, we have provided two values of lma, being thin leaves (0.07) or thick leaves (0.24). You could run this patch for just one species by defining just one lma value, or none if you just want to use the default trait values.
+
+```{r}
+results <- 
+  run_mypatch(traits=trait_matrix(c(0.07, 0.24), c("lma")))
+```
+
+The results from `run_scm_collect` in run_mypatch are in a 3D matrix state To convert this output to a more interpretable format, we can use tidy_patch to convert the results output into a tidy state.
+
+```{r}
+results_tidy <- 
+  results %>% 
+  tidy_patch()
+
+results_tidy
+```
+
+By pulling out the 'species' object from our list of tidied result outputs, we can plot the size distibution for each of the species in the patch. In this plot, each line represents the height over time of a characteristic originating at a given height (given by seed mass) at each step. Note that steps in this case are not equivalent to years, where the distance between steps is instead determined by the cohort spacing algorithm. We can take advantage of varying line transparency in ggplot2 to plot the relative log(Density) of each characteristic the size distribution of each species in the pathc. 
+
+```{r}
+results_tidy$species %>%
+  drop_na() %>%
```

Comment:
`drop_na()` is namepaced in `tidyr`, but we could re-export it if it's integral to the primary workflow.

Reply:
Let's see. For now it's in `imports` and loaded as a library for the vignette